lib/modules/datasource/nuget/v3.ts

```diff
@@ -242,3 +267,31 @@ export async function getReleases(
 
   return dep;
 }
+
+async function getSourceUrlFromNupkg(
```

Comment:
add the cache decorator to this function with a high cache time

Reply:
This is a function 😬

Reply:
why not using our cache decorator?

Reply:
You cannot use decorators on functions 😬

Reply:
declare it as class member of the datasource, there you can use the decorator

Reply:
The datasource is mostly forwarding to the v2 or v3 functions at the moment and I don't think it makes sense to add v3-specific logic there.

Should I send a PR to refactor v2 and v3 into classes first?

Reply:
I have sent https://github.com/renovatebot/renovate/pull/28117

Reply:
Switched to cache decorator with ca67215479147747d209ec929afbaa950cb0e6b5